### ISSUE TITLE
Add autocomplete for command labels

### ISSUE THREAD

**Issue by l4u** [atom/command-palette#36]

**Comment by izuzak** [comment]
@l4u Not sure I understand you're suggestion here -- can you provide an example and use that to clarify? Thanks!

**Comment by l4u** [comment]
Example: https://github.com/extesy/atom-text-manipulation/blob/f76597695586cffa07dc3ec6adeb767f4c86b0bc/menus/text-manipulation.cson#L31


Menu Item: Trim Leading/Trailing Spaces
Command: text-manipulation:whitespace-trim

I can type `whitespacetrim` in the command palette to search for that command.
But it would be great if we can also search a command by the label of the command at menu items.

**Comment by mnquintana** [comment]
@l4u I updated the title to better reflect what it seems like you're requesting - does it look alright to you?

**Comment by l4u** [comment]
Yeah it looks great.

**Comment by l4u** [comment]
I want to search commands by human readable labels. In my example the human readable label is `Trim Leading/Trailing Spaces` but the command is `text-manipulation:whitespace-trim`

**Comment by mnquintana** [comment]
Ah I see what you mean – so you'd want the commands to have completely different (ie. more descriptive) human readable labels, instead of just "humanizing" the command names?

**Comment by l4u** [comment]
Right. Currently we have human readable labels for menu items. But they are not searchable.

So we can add human readable labels to the commands, or we can add searchable menu items.

**Comment by idleberg** [comment]
I would also like to see this feature, since the current method doesn't work great for abbreviations or non-alphanumeric characters like a simple  dash (`-`). There are workarounds for both cases, but I'd rather have a clean labelling option like in the menus.

For backwards compatibility, I'd fallback to the command name if no label has been specified.
